--- vim73/src/auto/configure.xsetlocale~	2012-03-23 16:43:33.742180170 +0100
+++ vim73/src/auto/configure	2012-03-23 16:44:43.742259415 +0100
@@ -12021,66 +12021,6 @@ $as_echo "$vim_cv_memcpy_handles_overlap
 fi
 
 
-
-if test "$enable_multibyte" = "yes"; then
-  cflags_save=$CFLAGS
-  ldflags_save=$LDFLAGS
-  if test "x$x_includes" != "xNONE" ; then
-    CFLAGS="$CFLAGS -I$x_includes"
-    LDFLAGS="$X_LIBS $LDFLAGS -lX11"
-    { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether X_LOCALE needed" >&5
-$as_echo_n "checking whether X_LOCALE needed... " >&6; }
-    cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-#include <X11/Xlocale.h>
-int
-main ()
-{
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
-/* Override any GCC internal prototype to avoid an error.
-   Use char because int might match the return type of a GCC
-   builtin and then its argument prototype would still apply.  */
-#ifdef __cplusplus
-extern "C"
-#endif
-char _Xsetlocale ();
-int
-main ()
-{
-return _Xsetlocale ();
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_link "$LINENO"; then :
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-		$as_echo "#define X_LOCALE 1" >>confdefs.h
-
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-rm -f core conftest.err conftest.$ac_objext \
-    conftest$ac_exeext conftest.$ac_ext
-else
-  { $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-  fi
-  CFLAGS=$cflags_save
-  LDFLAGS=$ldflags_save
-fi
-
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for _xpg4_setrunelocale in -lxpg4" >&5
 $as_echo_n "checking for _xpg4_setrunelocale in -lxpg4... " >&6; }
 if test "${ac_cv_lib_xpg4__xpg4_setrunelocale+set}" = set; then :
--- vim73/src/configure.in.xsetlocale~	2012-03-23 16:43:26.101171514 +0100
+++ vim73/src/configure.in	2012-03-23 16:43:53.428202466 +0100
@@ -3389,26 +3389,6 @@ else
 fi
 
 
-dnl Check for multibyte locale functions
-dnl Find out if _Xsetlocale() is supported by libX11.
-dnl Check if X_LOCALE should be defined.
-
-if test "$enable_multibyte" = "yes"; then
-  cflags_save=$CFLAGS
-  ldflags_save=$LDFLAGS
-  if test "x$x_includes" != "xNONE" ; then
-    CFLAGS="$CFLAGS -I$x_includes"
-    LDFLAGS="$X_LIBS $LDFLAGS -lX11"
-    AC_MSG_CHECKING(whether X_LOCALE needed)
-    AC_TRY_COMPILE([#include <X11/Xlocale.h>],,
-	AC_TRY_LINK_FUNC([_Xsetlocale], [AC_MSG_RESULT(yes)
-		AC_DEFINE(X_LOCALE)], AC_MSG_RESULT(no)),
-	AC_MSG_RESULT(no))
-  fi
-  CFLAGS=$cflags_save
-  LDFLAGS=$ldflags_save
-fi
-
 AC_MSG_CHECKING([for locale directory])
 LOCALEDIR=`echo "${datadir}/locale"`
 AC_DEFINE_UNQUOTED(
