--- vim73/src/Makefile.icons_install~	2011-12-14 21:29:44.363793098 +0100
+++ vim73/src/Makefile	2011-12-14 21:30:26.261814468 +0100
@@ -2115,18 +2115,9 @@ ICON32PATH = $(DESTDIR)$(DATADIR)/icons/
 DESKTOPPATH = $(DESTDIR)$(DATADIR)/applications
 KDEPATH = $(HOME)/.kde/share/icons
 install-icons:
-	if test -d $(ICON48PATH) -a -w $(ICON48PATH) \
-		-a ! -f $(ICON48PATH)/gvim.png; then \
-	   $(INSTALL_DATA) $(SCRIPTSOURCE)/vim48x48.png $(ICON48PATH)/gvim.png; \
-	fi
-	if test -d $(ICON32PATH) -a -w $(ICON32PATH) \
-		-a ! -f $(ICON32PATH)/gvim.png; then \
-	   $(INSTALL_DATA) $(SCRIPTSOURCE)/vim32x32.png $(ICON32PATH)/gvim.png; \
-	fi
-	if test -d $(ICON16PATH) -a -w $(ICON16PATH) \
-		-a ! -f $(ICON16PATH)/gvim.png; then \
-	   $(INSTALL_DATA) $(SCRIPTSOURCE)/vim16x16.png $(ICON16PATH)/gvim.png; \
-	fi
+	install -m644 $(SCRIPTSOURCE)/vim48x48.png -D $(ICON48PATH)/gvim.png; \
+	install -m644 $(SCRIPTSOURCE)/vim32x32.png -D $(ICON32PATH)/gvim.png; \
+	install -m644 $(SCRIPTSOURCE)/vim16x16.png -D $(ICON16PATH)/gvim.png;
 	if test -d $(DESKTOPPATH) -a -w $(DESKTOPPATH); then \
 	   $(INSTALL_DATA) $(SCRIPTSOURCE)/vim.desktop \
 		$(SCRIPTSOURCE)/gvim.desktop \
