--- vim74/runtime/syntax/cpp.vim.cpp11~	2013-08-18 10:03:40.815065786 +0200
+++ vim74/runtime/syntax/cpp.vim	2013-08-18 10:07:10.565443546 +0200
@@ -35,9 +35,10 @@ syn keyword cppBoolean		true false
 
 " C++ 11 extensions
 if !exists("cpp_no_cpp11")
-  syn keyword cppType		override final
+  syn keyword cppType		override final nullptr_t tuple
   syn keyword cppExceptions	noexcept
-  syn keyword cppStorageClass	constexpr decltype
+  syn keyword cppStatement	static_assert
+  syn keyword cppStorageClass	constexpr decltype thread_local
   syn keyword cppConstant	nullptr
 endif
 
