--- vim74/src/config.h.in.localedir~	2013-02-26 14:18:19.000000000 +0100
+++ vim74/src/config.h.in	2013-08-18 10:08:03.748778663 +0200
@@ -413,6 +413,9 @@
 /* Define if we have dlopen() */
 #undef HAVE_DLOPEN
 
+/* Define path to localedir */
+#undef LOCALEDIR
+
 /* Define if we have dlsym() */
 #undef HAVE_DLSYM
 
--- vim74/src/configure.in.localedir~	2013-08-04 20:00:50.000000000 +0200
+++ vim74/src/configure.in	2013-08-18 10:08:03.749778650 +0200
@@ -3656,6 +3656,15 @@ if test "$enable_multibyte" = "yes"; the
   LDFLAGS=$ldflags_save
 fi
 
+AC_MSG_CHECKING([for locale directory])
+LOCALEDIR=`echo "${datadir}/locale"`
+AC_DEFINE_UNQUOTED(
+    [LOCALEDIR], ["$LOCALEDIR"],
+    [Full path to locale directory]
+)
+AC_SUBST(LOCALEDIR)
+AC_MSG_RESULT([$LOCALEDIR])
+
 dnl Link with xpg4, it is said to make Korean locale working
 AC_CHECK_LIB(xpg4, _xpg4_setrunelocale, [LIBS="$LIBS -lxpg4"],,)
 
--- vim74/src/main.c.localedir~	2013-07-03 12:36:49.000000000 +0200
+++ vim74/src/main.c	2013-08-18 10:08:03.750778638 +0200
@@ -1560,11 +1560,16 @@ init_locale()
 	/* expand_env() doesn't work yet, because chartab[] is not initialized
 	 * yet, call vim_getenv() directly */
 	p = vim_getenv((char_u *)"VIMRUNTIME", &mustfree);
+#ifndef LOCALEDIR
+
 	if (p != NULL && *p != NUL)
 	{
 	    vim_snprintf((char *)NameBuff, MAXPATHL, "%s/lang", p);
 	    bindtextdomain(VIMPACKAGE, (char *)NameBuff);
 	}
+#endif
+	bindtextdomain(VIMPACKAGE, LOCALEDIR);
+
 	if (mustfree)
 	    vim_free(p);
 	textdomain(VIMPACKAGE);
--- vim74/src/Makefile.localedir~	2013-08-18 10:08:03.712779113 +0200
+++ vim74/src/Makefile	2013-08-18 10:08:03.751778625 +0200
@@ -2136,7 +2136,7 @@ install-languages: languages $(DEST_LANG
 	-$(SHELL) ./installml.sh install "$(GUI_MAN_TARGETS)" \
 		$(DEST_MAN_RU_U) $(INSTALLMLARGS)
 	if test -n "$(MAKEMO)" -a -f $(PODIR)/Makefile; then \
-	   cd $(PODIR); $(MAKE) prefix=$(DESTDIR)$(prefix) LOCALEDIR=$(DEST_LANG) \
+	   cd $(PODIR); $(MAKE) prefix=$(DESTDIR)$(prefix) LOCALEDIR=$(DESTDIR)$(DATADIR)/locale \
 	   INSTALL_DATA=$(INSTALL_DATA) FILEMOD=$(FILEMOD) install; \
 	fi
 	if test -d $(LANGSOURCE); then \
--- vim74/src/po/Makefile.localedir~	2013-08-18 10:08:03.751778625 +0200
+++ vim74/src/po/Makefile	2013-08-18 10:08:38.709341596 +0200
@@ -157,11 +157,11 @@ install: $(MOFILES) $(MOCONVERTED)
 	for lang in $(LANGUAGES) $(CONVERTED); do \
 	  dir=$(LOCALEDIR)/$$lang/; \
 	  if test ! -x "$$dir"; then \
-	    mkdir $$dir; chmod 755 $$dir; \
+	    mkdir -p $$dir; chmod 755 $$dir; \
 	  fi; \
 	  dir=$(LOCALEDIR)/$$lang/LC_MESSAGES; \
 	  if test ! -x "$$dir"; then \
-	    mkdir $$dir; chmod 755 $$dir; \
+	    mkdir -p $$dir; chmod 755 $$dir; \
 	  fi; \
 	  if test -r $$lang.mo; then \
 	    $(INSTALL_DATA) $$lang.mo $$dir/$(PACKAGE).mo; \
