diff --git a/runtime/vimrc_example.vim b/runtime/vimrc_example.vim
index 7f5bae0..610a851 100644
--- a/runtime/vimrc_example.vim
+++ b/runtime/vimrc_example.vim
@@ -26,6 +26,11 @@ else
   endif
 endif
 
+" In many terminal emulators the mouse works just fine, thus enable it.
+if has('mouse')
+  set mouse=r
+endif
+
 if &t_Co > 2 || has("gui_running")
   " Switch on highlighting the last used search pattern.
   set hlsearch
