--- vim74/src/xxd/xxd.c.xxdloc~	2015-04-15 10:55:49.781922901 +0200
+++ vim74/src/xxd/xxd.c	2015-04-15 10:59:03.076275684 +0200
@@ -97,6 +97,7 @@
 #if __MWERKS__ && !defined(BEBOX)
 # include <unix.h>	/* for fdopen() on MAC */
 #endif
+#include <locale.h>
 
 #if defined(__BORLANDC__) && __BORLANDC__ <= 0x0410 && !defined(fileno)
 /* Missing define and prototype grabbed from the BC 4.0 <stdio.h> */
@@ -482,6 +483,11 @@ main(argc, argv)
   long length = -1, n = 0, seekoff = 0, displayoff = 0;
   static char l[LLEN+1];  /* static because it may be too big for stack */
   char *pp;
+  char *lang=getenv("LANG");
+  if(!lang) lang=getenv("LC_ALL");
+  if(!lang) lang=getenv("LC_CTYPE");
+  if(lang)
+         setlocale(LC_ALL, lang);
 
 #ifdef AMIGA
   /* This program doesn't work when started from the Workbench */
@@ -852,7 +858,7 @@ main(argc, argv)
 #else
 	  (e > 31 && e < 127)
 #endif
-	  ? e : '.';
+	  ? e : isalnum(e) ? e : '.';
       if (e)
 	nonzero++;
       n++;
